src/wp-admin/update-core.php

```diff
@@ -680,6 +680,13 @@ function list_theme_updates() {
 
 		$compat = '';
 
+		// Get the upgrade notice for the new theme version.
+		if ( isset( $theme->update['upgrade_notice'] ) ) {
+			$upgrade_notice = '<br/>' . strip_tags( $theme->update['upgrade_notice'] );
```

Comment:
```suggestion
			$upgrade_notice = $theme->update['upgrade_notice'];
```

Reply:
This is the identical code from the plugins on line 554

Reply:
The `update-core.php` page seems to be the only place in core that actually uses the `upgrade_notice` data.